src/Core/Framework/DataAbstractionLayer/Search/Criteria.php

```diff
@@ -109,13 +109,19 @@ class Criteria extends Struct
     protected $title;
 
     /**
-     * @param string[]|array<int, string[]> $ids
+     * @param string[]|array<int, string[]>|null $ids
      *
      * @throws InconsistentCriteriaIdsException
      */
-    public function __construct(array $ids = [])
+    public function __construct(?array $ids = null)
     {
-        if (\count($ids) > \count(array_filter($ids))) {
+        if ($ids === null) {
+            $this->ids = [];
+
+            return;
+        }
+
```

Comment:
To prepare for this change, we should trigger a deprecation warning on master. Could you create another PR?

Reply:
Will do!

Reply:
https://github.com/shopware/platform/pull/1421
Here it is